src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php

```diff
@@ -64,13 +65,19 @@ protected function doSendApi(SentMessage $sentMessage, Email $email, Envelope $e
             'body' => $body->bodyToIterable(),
         ]);
 
-        if (200 !== $response->getStatusCode()) {
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new HttpTransportException('Could not reach the remote Mailgun server.', $response, 0, $e);
+        }
+
+        if (200 !== $statusCode) {
             if ('application/json' === $response->getHeaders(false)['content-type'][0]) {
                 $result = $response->toArray(false);
```

Comment:
This one too could be wrapped.
Maybe wrapping should happen at a broader level?
or better: the call should be moved outside of the if, since it's needed below anyway

Reply:
Fixed